kolibri/content/utils/annotation.py

```diff
@@ -190,10 +190,9 @@ def recurse_availability_up_tree(channel_id):
         available_nodes = select([child.c.available]).where(
             and_(
                 child.c.available == True,  # noqa
-                child.c.level == level,
-                child.c.channel_id == channel_id,
+                ContentNodeTable.c.id == child.c.parent_id
             )
-        ).where(ContentNodeTable.c.id == child.c.parent_id)
```

Comment:
Oh interesting, SQLite doesn't optimize this into one single where clause?

Reply:
I think this might have been ok, but the additional superfluous queries I removed above definitely were an issue (I think it might have been doing many many JOINs to achieve them).